OneSignalSDK/onesignal/core/src/main/java/com/onesignal/common/events/EventProducer.kt

```diff
@@ -45,10 +45,9 @@ open class EventProducer<THandler> : IEventNotifier<THandler> {
      * @param callback The callback will be invoked for each subscribed handler, allowing you to call the handler.
      */
     fun fire(callback: (THandler) -> Unit) {
-        synchronized(subscribers) {
-            for (s in subscribers) {
-                callback(s)
-            }
+        val localList = subscribers.toList()
```

Comment:
Is `.toList()` thread safe? I looked into it a bit and it seems like this might not be due to the way Koltin makes a copy of it.

Reply:
@jkasten2 Good point! Technically, .toList() is not thread-safe, but the variable 'subscribers' is a synchronized list, preventing two threads from accessing it simultaneously. The only way to trigger the concurrent modification exception is to access the variable while it is being iterated over in a loop. Based on your comment, I've added a unit test that can throw a concurrent modification exception if any user subscribes to a handler that unsubscribes itself during the event. In this case, they are in the same thread, but unsubscribing is called while "subscribers" is iterating.

Reply:
Awesome, good test to have going forward in our unit test suite!